sw/device/silicon_creator/lib/drivers/pwrmgr.c

```diff
@@ -23,13 +24,25 @@ static_assert(
     PWRMGR_RESET_EN_EN_FIELD_WIDTH == 1,
     "RESET_EN field width changed, kAllResetsEnable may need to be updated");
 
+void pwrmgr_cdc_sync(uint32_t n) {
+  // We want to timeout if the CDC bit doesn't self clear.  It should take
+  // approximately 4 AON ticks to complete.  We wait 25% longer (5 ticks).
+  uint32_t cpu_cycle_timeout =
+      (uint32_t)kClockFreqCpuHz / (uint32_t)kClockFreqAonHz * 5;
+
+  while (n--) {
+    ibex_mcycle_zero();
+    abs_mmio_write32(kBase + PWRMGR_CFG_CDC_SYNC_REG_OFFSET, kSyncConfig);
```

Comment:
To avoid hardware lockups, we will probably have to perform read before write to ensure we don't write a 1 if this register is already set.

Reply:
Done.